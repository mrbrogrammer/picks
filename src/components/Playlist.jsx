import React, { Component } from 'react';
import '../App.css';


// Inline style for the counter text
let defaultStyle = {
    color: '#fff'
};


class Playlist extends Component {
    render() {
        let playlist = this.props.playlist
        let index = this.props.index;

        return (
            <div style={{...defaultStyle, backgroundColor: index % 2 ? '#C0C0C0' : '#808080', width: '25%', display: 'inline-block', margin: '10px', padding: '10px'}}>
                <img src={playlist.img ? playlist.img.url : ''} alt="" style={{width: '60px'}}/>
                <h3 style={{...defaultStyle, fontSize: '22px'}}>{playlist.name}</h3>
                <ul style={{marginTop: '10px', fontWeight: 'bold'}}>
                    {playlist.songs.map(song =>
                        <li style={{paddingTop: '2px'}}>{song.name}</li>
                    )}
                </ul>
            </div>
        );
    }
}


export default Playlist;