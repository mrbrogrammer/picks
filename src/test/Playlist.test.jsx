import { render } from '@testing-library/react';
import Playlist from '../components/Playlist';

describe('Test playlist Component', () => {
    const playlist = {
        name: 'Test Playlist',
        img: { url: 'https://example.com/image.jpg' },
        songs: [
            { name: 'Song 1' },
            { name: 'Song 2' }
        ]
    };

    it('Should render correctly with a playlist', () => {
        const result = render(<Playlist playlist={playlist} index={0} />);
        expect(result).toMatchSnapshot();
    });

    it('Should display the correct playlist name', () => {
        const { getByText } = render(<Playlist playlist={playlist} index={0} />);
        expect(getByText('Test Playlist')).toBeInTheDocument();
    });

    it('Should display the correct number of songs', () => {
        const { getAllByRole } = render(<Playlist playlist={playlist} index={0} />);
        expect(getAllByRole('listitem')).toHaveLength(2);
    });
});