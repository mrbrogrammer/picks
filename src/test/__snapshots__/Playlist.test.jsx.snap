// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test playlist Component Should render correctly with a playlist 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="color: rgb(255, 255, 255); background-color: rgb(128, 128, 128); width: 25%; display: inline-block; margin: 10px; padding: 10px;"
      >
        <img
          alt=""
          src="https://example.com/image.jpg"
          style="width: 60px;"
        />
        <h3
          style="color: rgb(255, 255, 255); font-size: 22px;"
        >
          Test Playlist
        </h3>
        <ul
          style="margin-top: 10px; font-weight: bold;"
        >
          <li
            style="padding-top: 2px;"
          >
            Song 1
          </li>
          <li
            style="padding-top: 2px;"
          >
            Song 2
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="color: rgb(255, 255, 255); background-color: rgb(128, 128, 128); width: 25%; display: inline-block; margin: 10px; padding: 10px;"
    >
      <img
        alt=""
        src="https://example.com/image.jpg"
        style="width: 60px;"
      />
      <h3
        style="color: rgb(255, 255, 255); font-size: 22px;"
      >
        Test Playlist
      </h3>
      <ul
        style="margin-top: 10px; font-weight: bold;"
      >
        <li
          style="padding-top: 2px;"
        >
          Song 1
        </li>
        <li
          style="padding-top: 2px;"
        >
          Song 2
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
